GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1053,10 +1053,7 @@ public bool ShowRemoteRef(IGitRef r)
         }
 
         [Browsable(false)]
-        public Task<bool> UnstagedChanges { get; private set; }
-
-        [Browsable(false)]
-        public Task<bool> StagedChanges { get; private set; }
+        public Task<IList<GitItemStatus>> ChangedFiles { get; private set; }
```

Comment:
Does it **have** to be of `Task<IList<GitItemStatus>>`? Can it be just of `IList<GitItemStatus>`?

Reply:
Will be changed

Reply:
Answered too soon: Task<> makes it possible to run this async, not wait in the calling procedure. The change works fine for me though, but I do not think it should be committed.

Reply:
It looks like this property is only used within `RevisionGrid.CheckUncommitedChanged()` method.
This means that:
1. it can be made private, and
2. it can be made a method
3. arguably it can be made async too - `GitGetCommitsCommandExited` kicks off process in async mode - that could probably be changed to `await`s....